crates/vsock/src/main.rs

```diff
@@ -43,45 +51,100 @@ struct VsockArgs {
     #[command(flatten)]
     param: Option<VsockParam>,
 
+    /// Configuration of the device corresponding to a guest in the form of comma separated key=value pairs.
+    /// The allowed keys are: guest_cid, socket, uds_path and tx_buffer_size
+    /// Example: --vm guest_cid=3,socket=/tmp/vhost3.socket,uds_path=/tmp/vm3.vsock,tx_buffer_size=65536
+    /// Multiple instances of this argument can be provided to configure devices for multiple guests.
+    #[arg(long, conflicts_with = "config", verbatim_doc_comment, value_parser = parse_vm_params)]
+    vm: Option<Vec<VsockConfig>>,
+
     /// Load from a given configuration file
     #[arg(long)]
     config: Option<String>,
 }
 
+fn parse_vm_params(s: &str) -> Result<VsockConfig> {
+    let mut guest_cid = None;
+    let mut socket = None;
+    let mut uds_path = None;
+    let mut tx_buffer_size = None;
+
+    for arg in s.trim().split(',') {
+        let mut parts = arg.split('=');
+        let key = parts.next().ok_or(Error::ArgsParse)?;
+        let val = parts.next().ok_or(Error::ArgsParse)?;
+
+        match key {
+            "guest_cid" | "guest-cid" => {
+                guest_cid = Some(val.parse().map_err(|_| Error::ArgsParse)?)
```

Comment:
I'd try to propagate the `ParseIntError` to our `Error::ArgsParse`.

Maybe we can define it as `ArgsParse(std::io::Error)` and then implement `From<ParseIntError> for std::io::Error`, since with `std::io::Error` we can also be more specific when we don't find the key.

Reply:
I don't understand the significance of using `std::io::Error here`. So for now, I am using a custom error type with descriptive fields

Reply:
Custom error type is even better ;-)